### ISSUE TITLE
MFS root shouldn't need to put the node in the DAG

### ISSUE THREAD

**Issue by schomatis** [ipfs/go-mfs#34]
See https://github.com/ipfs/go-ipfs/issues/5171.

**Comment by schomatis** [comment]
@nitishm Not really a beginner issue but take a look at it and tell me what you think, do you agree with the original statement?

**Comment by schomatis** [comment]
> Not sure I understand the context.

Yes, sorry, this isn't worded as an introductory problem to MFS but it seemed like the lowest hanging fruit among the group I'm working on at the moment, I'll do my best to guide you through it.

> I am assuming this is the `ipld.Node` which is the DAG root node

Yes.

> If the MFS root does not put the node

As mentioned in the original issue,

> the child should already have done this.

so, maybe this isn't happening at the moment, and we should move the "add to the DAG service" logic to the child which calls `closeChild` (now `updateChildEntry`).

**Comment by schomatis** [comment]
This may be easier to tackle if we simplify the entire call chain in https://github.com/ipfs/go-mfs/issues/39 (so basically let's bug @Stebalien to help us with that :)

**Comment by aschmahmann** [comment]
This repository is no longer maintained and has been copied over to [Boxo](https://github.com/ipfs/boxo).  In an effort to avoid noise and crippling in the Boxo repo from the weight of issues of the past, we are closing most issues and PRs in this repo.  Please feel free to open a new issue in Boxo (and reference this issue) if resolving this issue is still critical for unblocking or improving your usecase.

You can learn more in the [FAQs for the Boxo repo copying/consolidation effort](https://github.com/ipfs/boxo/wiki/Copied-or-Migrated-Repos-FAQ).
